ext/objspace/objspace_dump.c

```diff
@@ -679,6 +691,61 @@ objspace_dump(VALUE os, VALUE obj, VALUE output)
     return dump_result(&dc);
 }
 
+static void
+shape_i(rb_shape_t *shape, void *data)
+{
+    struct dump_config *dc = (struct dump_config *)data;
+
+    dump_append(dc, "{\"address\":");
+    dump_append_ref(dc, (VALUE)shape);
+
+    dump_append(dc, ", \"type\":\"SHAPE\", \"id\":");
+    dump_append_lu(dc, rb_shape_id(shape));
+
+    dump_append(dc, ", \"parent_id\":");
+    dump_append_lu(dc, shape->parent_id);
+
+    dump_append(dc, ", \"edges\":");
+    dump_append_sizet(dc, rb_shape_edges_count(shape));
+
+    dump_append(dc, ", \"shape_type\":");
+    switch(shape->type) {
+      case SHAPE_ROOT:
+        dump_append(dc, "\"ROOT\"");
+        break;
+      case SHAPE_IVAR:
+        dump_append(dc, "\"IVAR\"");
+        break;
+      case SHAPE_FROZEN:
+        dump_append(dc, "\"FROZEN\"");
+        break;
+      case SHAPE_CAPACITY_CHANGE:
+        dump_append(dc, "\"CAPACITY_CHANGE\"");
```

Comment:
Can we also append the capacity itself in this case, and the `SHAPE_INITIAL_CAPACITY` case? Something like:

```
dump_append(dc, ", \"capacity\":"); 
dump_append_sizet(dc, shape->capacity);
```

Reply:
No problem.